sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -1430,7 +1430,23 @@ class Dataset[T] private[sql](
    */
   @scala.annotation.varargs
   def select(cols: Column*): DataFrame = withPlan {
-    Project(cols.map(_.named), logicalPlan)
+    val untypedCols = cols.map {
+      case typedCol: TypedColumn[_, _] =>
```

Comment:
In order to not make a breaking change, we need to accept simple typed columns like `count`. @cloud-fan @HyukjinKwon